client/cli/src/commands/mod.rs

```diff
@@ -189,17 +188,24 @@ macro_rules! substrate_cli_subcommands {
 
 			fn network_config(
 				&self,
-				chain_spec: &::std::boxed::Box<dyn ::sc_service::ChainSpec>,
+				chain_spec: &std::boxed::Box<dyn sc_service::ChainSpec>,
 				is_dev: bool,
-				net_config_dir: ::std::path::PathBuf,
+				net_config_dir: std::path::PathBuf,
 				client_id: &str,
 				node_name: &str,
-				node_key: ::sc_service::config::NodeKeyConfig,
+				node_key: sc_service::config::NodeKeyConfig,
+				default_listen_port: u16,
```

Comment:
Is this parameter actually a default value or is this something that is being configured by the user somehow and then actually used to construct an instance of some type?

Reply:
The default ports are hardcoded in sc-cli but during runtime the user can override them. Cumulus has a third way to this approach: it overrides the default ports so the user can still override at runtime but the default value would be different.

This is because cumulus has 2 nodes running.